src/c4/QuoWrapper.hh

```diff
@@ -0,0 +1,110 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   c4/QuoWrapper.hh
+ * \author Kelly Thompson
+ * \date   Friday, Nov 29, 2019, 18:56 pm
+ * \brief  A singleton, light-weight wrapper around Quo (github.com/lanl/libquo)
+ * \note   Copyright (C) 2019 Triad National Security, LLC.,
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef rtt_c4_quowrapper_hh
+#define rtt_c4_quowrapper_hh
+
+#include "c4/config.h"
+#include "ds++/Assert.hh"
+#include <cstdint>
+#include <string>
+
+namespace rtt_c4 {
+
+//============================================================================//
+/*!
+ * \class QuoWrapper
+ * \brief A singlton that is a light-weight wrapper around the Quo API
+ *        (https://gitlab.lanl.gov/lanl/libquo).
+ */
+//============================================================================//
+
+class QuoWrapper {
+public:
+  //! Destructor (free the quo context)
+  ~QuoWrapper() {
+#ifdef HAVE_LIBQUO
+    QUO_free(quo);
```

Comment:
Inject this logic into `rtt_c4::finalize()` since MPI must remain in a valid state until after ``QUO_free` is called.

Reply:
Fixed in 8e17a4b1fee52afc5efb831b6b396f366a4656ce.